feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/data/signer/proxy/ProxiedSigner.kt

```diff
@@ -78,9 +94,32 @@ class ProxiedSigner(
         return signerProvider.signerFor(proxyMetaAccount)
     }
 
-    private suspend fun modifyPayload(proxyMetaAccount: MetaAccount, payload: SignerPayloadExtrinsic): SignerPayloadExtrinsic {
-        val chain = chainRegistry.getChain(payload.chainId)
+    private suspend fun validateExtrinsic(signedExtrinsic: SignerPayloadExtrinsic, chain: Chain) {
+        val proxyAccount = accountRepository.getLastProxyAccountFor(proxiedMetaAccount.id) ?: throw IllegalStateException("Proxy account is not found")
+        val proxyAccountId = proxyAccount.accountIdIn(chain) ?: throw IllegalStateException("Proxy account is not found")
+
+        val callInstance = signedExtrinsic.call.toCallInstance() ?: signingNotSupported()
+
+        val validationPayload = ProxiedExtrinsicValidationPayload(
+            proxyAccount,
+            proxyAccountId,
+            ChainWithAsset(chain, chain.commissionAsset),
+            callInstance.call
+        )
```

Comment:
Unfortunately we should to pass call instance instead of calculating fee here, because we can't inject ExtrinsicService here due to cyclical dependencies